### PR TITLE
[stable/drone] Drone 2.0.0 rc 8 - Creates source-control secret

### PR DESCRIPTION
**Pull request by zakkg3**
#### What this PR does / why we need it:

The chart is failing if you don't create in advance the provider's secret, It tries to read it from `<fullName>-source-control` but the secret its not there, it does not create it for you if you want to. 
This pr automate the secret creation by reading it form the new Secret field and creates "ClientSecretKey: Secret", while keeping the option to use a pre created secret, as it using the same key (ClientSecretKey).

/edit I also fixed the note, it was pointing the proxy to port 8000, now to 80. working fine :)

#### Special notes for your reviewer:
@max-rocket-internet @paulczar @christian-roggia 
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by zakkg3** [comment]
Hey guys @prydonius @christian-roggia what do you think?

**Comment by zakkg3** [comment]
@cpanato @pedrovaz  I would like some feedback :)

File: stable/drone/Chart.yaml [hunk 1]
```diff
@@ -19,3 +19,5 @@ maintainers:
   email: christian.roggia@gmail.com
 - name: paulczar
   email: username.taken@gmail.com
+- name: zakkg3
```
Comment (cpanato) [inline stable/drone/Chart.yaml]: why adding yourself here?

**Comment by cpanato** [comment]
@paulczar I know you did a lot of work here, any comments?

File: stable/drone/Chart.yaml (hunk 1 above)
Comment (zakkg3) [reply stable/drone/Chart.yaml]: Mistake, deleted. solved :) Ty

**Comment by zakkg3** [comment]
Version bumped to rc.9 as yesterday rc.8 was merged

File: stable/drone/values.yaml [hunk 2]
```diff
@@ -80,15 +80,17 @@ sourceControl:
   secret:
   ## Fill in the correct values for your chosen source control provider
   ## Any key in this list with the suffix `Key` will be fetched from the
-  ## secret named above, if not provided the secret will default to
-  ## `<fullName>-source-control`
+  ## secret named above (you need to create in advance), if not provided the secret will default to
+  ## `<fullName>-source-control` and use the "ClientSecretKey:""secret" fields to create it. (if then, be awere to not share shis file with your password on it)
   github:
```
Comment (christian-roggia) [inline stable/drone/values.yaml:86]: This part of the documentation is very confusing, can you rewrite it in a clearer way?

File: stable/drone/values.yaml [hunk 3]
```diff
@@ -80,15 +80,17 @@ sourceControl:
...
     clientID:
     clientSecretKey: clientSecret
+    Secret:
     server: https://github.com
   gitlab:
     clientID:
     clientSecretKey: clientSecret
+    Secret:
```
Comment (christian-roggia) [inline stable/drone/values.yaml]: Normally camel case for variables is used with a lowercase first letter. I would also use a better naming convention like `secretValue` or `clientSecretValue`.

File: stable/drone/templates/deployment-server.yaml [hunk 4]
```diff
@@ -14,7 +14,7 @@ spec:
   template:
     metadata:
       annotations:
-        checksum/secrets: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        checksum/secrets: {{ include (print $.Template.BasePath "/secret-shared-rpc.yaml") . | sha256sum }}
```
Comment (christian-roggia) [inline stable/drone/templates/deployment-server.yaml]: Wouldn't be also an option to merge the secrets in a single file? I wonder if having multiple secrets is really required. There is also no checksum on the new secret.

**Comment by zakkg3** [comment]
Thanks, @christian-roggia and @max-rocket-internet for the feedback. 
Changelog:
* Merged secrets file back to original "secrets.yaml" with checksum
* Updated documentation within a clearer way
* Rename .Values.Secret -> clientSecretValue

Let me know if there is something else we can improve here.
updated my deployment, Happy Helming!

**Review by max-rocket-internet — APPROVED** [review]

**Comment by cpanato** [comment]
/approve
will defer the lgtm to @christian-roggia

**Review by cpanato — APPROVED** [review]

**Comment by zakkg3** [comment]
@christian-roggia 
any comments on this? Thx!!!

**Pull request merged**